### ISSUE TITLE
Out of memory error

### ISSUE THREAD

**Issue by sunwoo76** [microsoft/CoCosNet#21]
What is the minimum capacity of memory for training with the batch_size=1?

I implemented a 2060super GPU on my computer, but this error happened.

**Comment by zhangmozhe** [comment]
Possibly you can choose pytorch 1.6 and adopt mixed-precision training for memory-efficient training.

**Comment by panzhang0212** [comment]
16G is ok, and 32G is better with a larger batch size

**Comment by sunwoo76** [comment]
@panzhang0212 thank you.  I took a typo in my comment. I implemented  GPU2070 super not 2060super and its memory is 8G.
Will not 8GB memory size occur out of memory error with batch size 1?

**Comment by mlxht990720** [comment]
I have a same problem which is out of memory error , I use GPU2080Ti and its memory is 10GB, though I change batch_size to 2, out of memory also happened .

**Comment by myway0101** [comment]
@mlxht990720 Did you try with batch_size = 1 using 2080Ti?

**Comment by mlxht990720** [comment]
Yes. But it's still out of memory. Maybe I need to use more gpus.🥲



发自我的iPhone


------------------ Original ------------------
From: myway0101 ***@***.***&gt;
Date: Fri,Apr 9,2021 3:37 PM
To: microsoft/CoCosNet ***@***.***&gt;
Cc: mlxht990720 ***@***.***&gt;, Mention ***@***.***&gt;
Subject: Re: [microsoft/CoCosNet] Out of memory error (#21)

**Comment by myway0101** [comment]
Any idea to decrease memory cost in training?

**Comment by zhangmozhe** [comment]
> Any idea to decrease memory cost in training?

You can try to use mixed precision training and gradient checkpointing. Further, you can try https://github.com/facebookresearch/fairscale

**Comment by muzili028** [comment]
maybe you can resize the image into 256*256
